src/Components/Web/src/Forms/InputLargeTextArea/InputLargeTextArea.cs

```diff
@@ -0,0 +1,89 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// A multiline input component for editing large <see cref="string"/> values, supports async
+    /// content access without binding nor validations.
+    /// </summary>
+    public class InputLargeTextArea : ComponentBase, IInputLargeTextAreaJsCallbacks, IDisposable
+    {
+        private ElementReference _inputLargeTextAreaElement;
+
+        private InputLargeTextAreaJsCallbacksRelay? _jsCallbacksRelay;
+
+        [Inject]
+        private IJSRuntime JSRuntime { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the event callback that will be invoked when the textarea content changes.
+        /// </summary>
+        [Parameter]
+        public EventCallback<InputLargeTextAreaChangeEventArgs> OnChange { get; set; }
+
+        /// <summary>
+        /// Gets or sets a collection of additional attributes that will be applied to the input element.
+        /// </summary>
+        [Parameter(CaptureUnmatchedValues = true)]
+        public IDictionary<string, object>? AdditionalAttributes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the associated <see cref="ElementReference"/>.
+        /// <para>
+        /// May be <see langword="null"/> if accessed before the component is rendered.
+        /// </para>
+        /// </summary>
+        [DisallowNull]
+        public ElementReference? Element
+        {
+            get => _inputLargeTextAreaElement;
+            protected set => _inputLargeTextAreaElement = value!.Value;
+        }
+
+        /// <inheritdoc/>
+        protected override async Task OnAfterRenderAsync(bool firstRender)
+        {
+            if (firstRender)
+            {
+                _jsCallbacksRelay = new InputLargeTextAreaJsCallbacksRelay(this);
+                await JSRuntime.InvokeVoidAsync(InputLargeTextAreaInterop.Init, _jsCallbacksRelay.DotNetReference, _inputLargeTextAreaElement);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "textarea");
+            builder.AddMultipleAttributes(1, AdditionalAttributes);
+            builder.AddElementReferenceCapture(2, elementReference => _inputLargeTextAreaElement = elementReference);
+            builder.CloseElement();
+        }
+
+        Task IInputLargeTextAreaJsCallbacks.NotifyChange(int length)
+            => OnChange.InvokeAsync(new InputLargeTextAreaChangeEventArgs(length));
+
+        /// <summary>
+        /// Retrieves the textarea value asyncronously.
+        /// </summary>
+        /// <returns>The string value of the textarea.</returns>
+        public ValueTask<string> GetTextAsync()
```

Comment:
Should these need to include cancelation tokens? It's odd to see async APIs that do not.

Reply:
Would it be weird to have these APIs support lines of text? e.g.

```C#
public ValueTask<IEnumerable<string>> GetTextAsync(CancellationToken cts)
public ValueTask SetTextAsync(IEnumerable<string> textLines, CancelationToken cts)
```

The reason I say this is that at over 64k chars, this component will have really poor memory characteristics. Having the input chunked would mitigate some of that.

Reply:
> Should these need to include cancelation tokens? It's odd to see async APIs that do not.

Sure, I think that'd be a great addition.



> Would it be weird to have these APIs support lines of text? e.g.
> 
> ```cs
> public ValueTask<IEnumerable<string>> GetTextAsync(CancellationToken cts)
> public ValueTask SetTextAsync(IEnumerable<string> textLines, CancelationToken cts)
> ```
> 
> The reason I say this is that at over 64k chars, this component will have really poor memory characteristics. Having the input chunked would mitigate some of that.

As you point out below, we don't support arbitrarily large text. As it stands, this component is still constrained by the SignalR message size limit (32kb here). As a part of the proposal I included an alternate streaming based API, but I'm unsure if this is something we want to pursue now or save for later when users request it:

```c#
        /// <summary>
        /// Retrieves the textarea value asynchronously.
        /// </summary>
        /// <returns>The string value of the textarea.</returns>
        public ValueTask<Stream> GetTextAsync();

        /// <summary>
        /// Sets the textarea value asynchronously.
        /// </summary>
        /// <param name="newValue">The new content to set for the textarea.</param>
        public ValueTask SetTextAsync(Stream newValue);
```